ui/app/src/components/Elements/SifSelect.vue

```diff
@@ -0,0 +1,143 @@
+<template>
+  <div class="sif-select">
+    <div class="sif-select__activator" @click="openMenu">
+      <SifButton primary block v-if="!selected">Select</SifButton>
+      <SifButton secondary block v-else>{{ selected.name }}</SifButton>
+    </div>
+    <div class="sif-select__wrapper" v-if="isOpen" @click="closeMenu">
+      <div class="sif-select__content" @click.stop>
+        <div class="sif-select__close" @click="closeMenu">&times;</div>
+        <div class="sif-select__header">
+          <h3 class="sif-select__title">Select a token</h3>
+          <SifInput gold placeholder="Search name or paste address" />
+          <h4 class="sif-select__list-title">Token Name</h4>
+        </div>
+        <div class="sif-select__body">
+          <div 
+            class="sif-select__option" 
+            v-for="(token, index) in tokens" 
+            :key="index"
+            @click="selectOption($event, token)">
+            <span>{{ token.name }}</span>
+          </div>
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
```

Comment:
We probably need to work out what our TypeScript policy is and have a discussion around that.